database/gorm/query.go

```diff
@@ -780,6 +866,82 @@ func (r *QueryImpl) Where(query any, args ...any) ormcontract.Query {
 	return r.setConditions(conditions)
 }
 
+func (r *QueryImpl) WhereHas(relation string, callback func(query ormcontract.Query) (ormcontract.Query, error), args ...any) ormcontract.Query {
```

Comment:
This method needs to be implemented like other methods, saving the conditions to r.conditions first, and adding a new `buildWhereHas` method in the `buildConditions` method.